testnet/stacks-node/src/mockamoto/tests.rs

```diff
@@ -126,3 +127,156 @@ fn observe_100_blocks() {
+    let submitter_addr = to_addr(&submitter_sk);
+    conf.add_initial_balance(submitter_addr.to_string(), 1_000);
+
+    test_observer::spawn();
+    let observer_port = test_observer::EVENT_OBSERVER_PORT;
+    conf.events_observers.push(EventObserverConfig {
+        endpoint: format!("localhost:{observer_port}"),
+        events_keys: vec![EventKeyType::AnyEvent],
+    });
+
+    let mut mockamoto = MockamotoNode::new(&conf).unwrap();
+
+    let globals = mockamoto.globals.clone();
+
+    let mut mempool = PeerThread::connect_mempool_db(&conf);
+    let (mut chainstate, _) = StacksChainState::open(
+        conf.is_mainnet(),
+        conf.burnchain.chain_id,
+        &conf.get_chainstate_path_str(),
+        None,
+    )
+    .unwrap();
+    let burnchain = conf.get_burnchain();
+    let sortdb = burnchain.open_sortition_db(true).unwrap();
+    let sortition_tip = SortitionDB::get_canonical_burn_chain_tip(mockamoto.sortdb.conn()).unwrap();
+
+    let start = Instant::now();
+    // Get a reward cycle to compare against
+    let reward_cycle = mockamoto
+        .sortdb
+        .pox_constants
+        .block_height_to_reward_cycle(
+            mockamoto.sortdb.first_block_height,
+            sortition_tip.block_height,
+        )
+        .expect(
+            format!(
+                "Failed to determine reward cycle of block height: {}",
+                sortition_tip.block_height
+            )
+            .as_str(),
+        );
+
+    let node_thread = thread::Builder::new()
+        .name("mockamoto-main".into())
+        .spawn(move || {
+            mockamoto.run();
+            let aggregate_key_block_header = NakamotoChainState::get_canonical_block_header(
+                mockamoto.chainstate.db(),
+                &mockamoto.sortdb,
+            )
+            .unwrap()
+            .unwrap();
+            // Get the aggregate public key to later verify that it was set correctly
+            mockamoto
+                .chainstate
+                .get_aggregate_public_key_pox_4(
+                    &mockamoto.sortdb,
+                    &aggregate_key_block_header.index_block_hash(),
+                    reward_cycle,
+                )
+                .unwrap()
+        })
+        .expect("FATAL: failed to start mockamoto main thread");
+
+    // Create a "set-aggregate-public-key" tx to verify it sets correctly
+    let mut rng = OsRng::default();
+    let x = Scalar::random(&mut rng);
+    let random_key = Point::from(x);
+
+    let aggregate_public_key = Value::buff_from(random_key.compress().data.to_vec())
+        .expect("Failed to serialize aggregate public key");
+    let aggregate_tx = make_contract_call(
+        &submitter_sk,
+        0,
+        10,
+        &boot_code_addr(false),
+        POX_4_NAME,
+        "set-aggregate-public-key",
+        &[Value::UInt(u128::from(reward_cycle)), aggregate_public_key],
+    );
+    let aggregate_tx_hex = format!("0x{}", to_hex(&aggregate_tx));
```

Comment:
Can this test run without this `contract-call`, and instead line 254 asserting that `aggregate_key == node.self_signer.aggregate_public_key`? That way, it's checking that the mockamoto setup code itself is successfully setting the aggregate key.

Reply:
I will update to do both! Cause then it will test but contract call to set as well as code logic to init it correctly